src/test/lib/optimizer/optimizer_test.cpp

```diff
@@ -331,4 +337,43 @@ TEST_F(OptimizerTest, OptimizesSubqueriesExactlyOnce) {
   }
 }
 
+TEST_F(OptimizerTest, CheckTrueCacheablility) {
```

Comment:
i think this test name (as well as the following) are a nit misleading. Also, add a comment on what they test.